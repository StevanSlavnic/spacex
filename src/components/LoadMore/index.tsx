import { InView } from "react-intersection-observer";

export default function LoadMore({
  hasNextPage,
  onLoadMore,
}: {
  hasNextPage: boolean;
  onLoadMore: () => void;
}) {
  return (
    <>
      <InView
        as={"div"}
        onChange={(inView) => {
          if (hasNextPage && inView) {
            onLoadMore();
          }
        }}
      >
        <div className="flex justify-center items-center"></div>
      </InView>
    </>
  );
}
