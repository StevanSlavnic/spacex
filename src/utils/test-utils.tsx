import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { SearchProvider } from "../context/SearchContext";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";

const queryClient = new QueryClient({});

export const AllTheProviders = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  return (
    <QueryClientProvider client={queryClient}>
      <SearchProvider>{children}</SearchProvider>
    </QueryClientProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
