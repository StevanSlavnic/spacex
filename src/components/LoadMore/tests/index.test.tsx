import { render, waitFor } from "@testing-library/react";
import "react-intersection-observer/test-utils";
import "@testing-library/jest-dom";
import LoadMore from "..";

describe("LoadMore Component", () => {
  const intersectionObserverMock = () => ({
    observe: () => null,
  });
  window.IntersectionObserver = jest
    .fn()
    .mockImplementation(intersectionObserverMock);
  const debounceLoadMoreArticles = jest.fn();
  it("should render without crashing", () => {
    const { container } = render(
      <LoadMore hasNextPage={true} onLoadMore={debounceLoadMoreArticles} />
    );
    expect(container).toMatchSnapshot();
  });

  it("should call debounceLoadMoreArticles when in view and hasNextPage is true", () => {
    render(
      <LoadMore hasNextPage={true} onLoadMore={debounceLoadMoreArticles} />
    );

    waitFor(() => {
      expect(debounceLoadMoreArticles).toHaveBeenCalled();
    });
  });

  it("should not call debounceLoadMoreArticles when not in view", () => {
    const debounceLoadMoreArticles = jest.fn();
    render(
      <LoadMore hasNextPage={true} onLoadMore={debounceLoadMoreArticles} />
    );

    expect(debounceLoadMoreArticles).not.toHaveBeenCalled();
  });

  it("should not call debounceLoadMoreArticles when hasNextPage is false", () => {
    const debounceLoadMoreArticles = jest.fn();
    render(
      <LoadMore hasNextPage={false} onLoadMore={debounceLoadMoreArticles} />
    );

    expect(debounceLoadMoreArticles).not.toHaveBeenCalled();
  });
});
