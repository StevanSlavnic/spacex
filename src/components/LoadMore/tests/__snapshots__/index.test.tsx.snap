// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadMore Component should render without crashing 1`] = `
<div>
  <div>
    <div
      class="flex justify-center items-center"
    />
  </div>
</div>
`;
